### PR TITLE
Fix help for financial acls.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Follow up fix on https://github.com/civicrm/civicrm-core/pull/13047 to get help text on financial acls working on contribution settings page (addresses temporary regression in master / unreleased code)

Before
----------------------------------------
screenshot 2018-11-05 09 37 12

After
----------------------------------------
screenshot 2018-11-05 09 40 56

Technical Details
----------------------------------------
**To test this make sure you do a cv flush or similar since setting metadata is cached** - also the help text is slow to load locally - not sure why but don't give up :-) (this is not a new slowness)

@mattwire @pradpnayak this is kind of exciting

I've done this by using the field template that is used by entity forms & adding
help to the setting spec (which I'll document).

I noted in this that entityTrait used a different field key for the
documentation link - literally documentation_link whereas I had used field_link
so I fixed settings to match

Comments
----------------------------------------
Remaining todo list

- ~~address the help text issue~~
- see if I can create the generic form per https://lab.civicrm.org/dev/core/issues/495 -UPDATE - see https://github.com/civicrm/civicrm-core/pull/13054
- update docs
- look at handling casing in a noisy but non-fail-y way
- if you want to put your PR up as an improvement I'll review & merge that
- audit settings / preferences forms to check for any failures
- email dev & request people specifically test settings forms in the rc

### CODE REVIEW

File: settings/Multisite.setting.php [hunk 1]
```diff
@@ -48,7 +48,7 @@
     'is_domain' => 1,
     'is_contact' => 0,
     'description' => ts('Make CiviCRM aware of multiple domains. You should configure a domain group if enabled'),
-    'help_link' => ['page' => 'Multi Site Installation', 'resource' => 'wiki'],
+    'documentation_link' => ['page' => 'Multi Site Installation', 'resource' => 'wiki'],
```
Comment (seamuslee001) [inline settings/Multisite.setting.php]: @eileenmcnaughton these are giving off 500 errors linking to docs.civicrm.org not the old wiki

File: settings/Event.setting.php [hunk 2]
```diff
@@ -48,7 +48,7 @@
     'is_contact' => 0,
     'description' => ts('This feature allows users to register for more than one event at a time. When enabled, users will add event(s) to a "cart" and then pay for them all at once. Enabling this setting will affect online registration for all active events. The code is an alpha state, and you will potentially need to have developer resources to debug and fix sections of the codebase while testing and deploying it'),
     'help_text' => '',
-    'help_link' => ['page' => 'CiviEvent Cart Checkout', 'resource' => 'wiki'],
+    'documentation_link' => ['page' => 'CiviEvent Cart Checkout', 'resource' => 'wiki'],
```
Comment (seamuslee001) [inline settings/Event.setting.php]: @eileenmcnaughton and here too

**Comment by seamuslee001** [comment]
(*CiviCRM Review Template WORD-1.1*)



* ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) Pass
* ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) Pass
* ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) Pass
* ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) Pass
* ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) Pass
* ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) Mixed results not pass - help link works for Fin Acls but documentation links now going to 500 errors
* ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) Pass
* ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) Pass

**Comment by eileenmcnaughton** [comment]
@seamuslee001 fixed

File: templates/CRM/Core/Form/Field.tpl [hunk 3]
```diff
@@ -35,6 +35,6 @@
   </td>
   <td>{if $form.$fieldName.html}{if $fieldSpec.formatter === 'crmMoney'}{$form.$fieldName.html|crmMoney}{else}{$form.$fieldName.html}{/if}{else}{$fieldSpec.place_holder}{/if}<br />
     {if $fieldSpec.description}<span class="description">{$fieldSpec.description}</span>{/if}
-    {if $fieldSpec.documentation_link}{docURL page=$fieldSpec.documentation_link.page}{/if}
+    {if $fieldSpec.documentation_lCink}{docURL page=$fieldSpec.documentation_link.page resource=$fieldSpec.documentation_link.resource}{/if}
```
Comment (seamuslee001) [inline templates/CRM/Core/Form/Field.tpl]: @eileenmcnaughton typo

Comment (eileenmcnaughton) [reply templates/CRM/Core/Form/Field.tpl]: argh it must have recorded a key stroke in the wrong window

**Comment by seamuslee001** [comment]
After re-testing (manually fixing the typo) I can confirm all links work and the ACL help is correct adding merge on pass

**Comment by eileenmcnaughton** [comment]
thanks @seamuslee001

File: settings/Multisite.setting.php (hunk 1 above)
Comment (seamuslee001) [reply settings/Multisite.setting.php]: with the changes to `templates/CRM/Core/Form/Field.tpl` this is now resolved

**Pull request merged**